### ISSUE TITLE
Using Browser Integration from keepassxc-cli

### ISSUE THREAD

**Issue by bendem** [keepassxreboot/keepassxc#5069]
## Summary

When using the cli, instead of specifying a path and entering your password every single time you interact with a database, connect to the running keepassxc using the browser integration.

## Examples

```bash
keepassxc-cli show --from-open-databases -a Password 'Production/devops/gitea-production'

# we use ansible-vault to store secrets, passwords are in keepass
ansible-vault show --vault-id gitea-production@kpxc-prod-client some/file.vault
ansible-playbook -l prod \
  --vault-id gitea-production@kpxc-prod-client \
  --vault-id postgres-production@kpxc-prod-client \
  playbooks/gitea/setup.yml
```

```bash
#!/bin/bash

set -euo pipefail
set -x

readonly KP_CLI=${KP_CLI:-"$USERPROFILE/scoop/apps/keepassxc/current/keepassxc-cli.exe"}
readonly KP_ROOT=${KP_ROOT:-'Production/devops/gitea-production'}

"$KP_CLI" show --from-open-databases -a Password "$KP_DATABASE" "$KP_ROOT/$2"
```

## Context

We use ansible and ansible-vaults to encrypt secrets, to unlock those secrets we have a script we give to ansible that fetches passwords from the keepass database. Since the cli touches the database file directly, we have to unlock the database every single time.

I saw the open feature, but this is unusable for vaults since the script is run once per password and even if it worked, it would still require the user to enter their password at least once per run.

## Bonus

If all commands support the new option, the complexity of having to maintain a REPL can be entirely removed from the code.

**Comment by bendem** [comment]
This is kind of related to #4513 which I didn't find at first, but I feel more strongly about the solution required. Browser integration is already there, it works, I don't see any reason why the cli couldn't be yet another client of it.

As for the key, it can be stored in linux with libsecret, mac's keychain and windows probably has a thing similar. It will actually be more secure than browser's storage where it's just a file or a db in plaintext in the user's profile directory.

**Comment by droidmonkey** [comment]
Good idea

**Comment by Frederick888** [comment]
Shameless advertisement: https://github.com/Frederick888/git-credential-keepassxc#scripting

**Comment by droidmonkey** [comment]
Yes!

**Comment by bendem** [comment]
> Shameless advertisement: https://github.com/Frederick888/git-credential-keepassxc#scripting

I wish, but that doesn't work on windows:

```
bendem> git-credential-keepassxc.exe configure
Jul 15 13:57:36.999 ERRO Le fichier spécifié est introuvable. (os error 2), Caused by: N/A, Message: Le fichier spécifié est introuvable. (os error 2)
[98.99 ms]
```

**Comment by Frederick888** [comment]
@bendem 

1. I don't speak French(?)
0. `-vv` to enable verbose logs
0. Even better, do a debug build and then `-vvv`
0. You should file an issue in my repo instead

**Comment by nodiscc** [comment]
I am after basically the same thing,e.g. `keepassxc-cli show --use-proxy --show-protected -a Password /path/to/my/database.kdbx 'my-ansible-vault-password'`

I posted a duplicate issue by mistake in https://github.com/keepassxreboot/keepassxc/issues/10238, but it also includes a workaround using https://github.com/hargoniX/keepassxc-proxy-client

The browser integration mechanism seems to be a good fit, as it provides fine-grained access to different databases/entries.

> As for the key, it can be stored in linux with libsecret, mac's keychain and windows probably has a thing similar.

My workaround stores the native messaging key in plain text on disk, but +1 :+1: to this idea.

**Comment by jfsanchez** [comment]
This feature would be wonderful.

I would love to use keepassxc to store the credentials and path of the webdav/ftp/whatever where it would sync its own database. Even if I need to memorice that too just in case, at least sync data would be more protected and I could keep the config updated among devices and I could share just part of the DB (autoopen and export to another DB). Merge could also be easily scriptable and the config stored inside too.

**Comment by Lippiece** [comment]
> Shameless advertisement: [Frederick888/git-credential-keepassxc#scripting](https://github.com/Frederick888/git-credential-keepassxc?rgh-link-date=2020-07-14T23%3A29%3A15.000Z#scripting)

This is really an answer. See an [example](https://github.com/Frederick888/git-credential-keepassxc/discussions/111) of how to use it like `pass`. Thanks, @Frederick888.

**Comment by jfsanchez** [comment]
Yes, thank you I read it but I don't like the idea of compiling another program. It would be great to have it integrated and working on different computers. I suppose I could integrate it inside the AppImage for easy portability but having to make it every time an update is released is not appealing.

I am still figuring out what's the best way to sync some DB among computers.
